src/Workspaces/MSBuild/BuildHost/Build/ProjectBuildManager.cs

```diff
@@ -95,6 +96,47 @@ private ImmutableDictionary<string, string> AllGlobalProperties
 
             using var stream = FileUtilities.OpenAsyncRead(path);
             using var readStream = await SerializableBytes.CreateReadableStreamAsync(stream, cancellationToken).ConfigureAwait(false);
+            return LoadProjectCore(path, readStream, projectCollection, log);
+        }
+        catch (Exception e)
+        {
+            log.Add(e, path);
+            return (project: null, log);
+        }
+    }
+
+    public (MSB.Evaluation.Project? project, DiagnosticLog log) LoadProject(string path, Stream readStream)
```

Comment:
I think we can reuse more between the on-disk and in-memory paths, taking a look.